libfwupd/fwupd-client.c

```diff
@@ -491,6 +597,7 @@ fwupd_client_connect_get_proxy_cb (GObject *source,
 		fwupd_client_set_host_security_id (self, g_variant_get_string (val, NULL));
 
 	/* success */
+	fwupd_client_set_status (self, FWUPD_STATUS_IDLE);
```

Comment:
This change looks unrelated

Reply:
It's actually completely related; it causes a property change event which I'm using in the context test to check the GSource is working as we expect. I guess the proper fix is to do something like:

    val = g_dbus_proxy_get_cached_property (proxy, "Status");
    fwupd_client_set_status (self, g_variant_get_uint32 (val));

as I think that's set after the proxy is set up.

Reply:
Yes, this works -- Status was missing completely!

Reply:
Perhaps it should be a separate commit, with some explanation along the lines of what you’ve just said in the commit message.

Reply:
I did exactly that :)

Reply:
Sorry, I think I was reviewing a stale version of the branch at that point! :partying_face: